pkg/operator/ceph/object/spec.go

```diff
@@ -245,11 +245,18 @@ func (c *clusterConfig) reconcileService(cephObjectStore *cephv1.CephObjectStore
 		if !kerrors.IsAlreadyExists(err) {
 			return "", errors.Wrapf(err, "failed to create ceph object store service")
 		}
-		svc, err = c.context.Clientset.CoreV1().Services(cephObjectStore.Namespace).Get(instanceName(cephObjectStore.Name), metav1.GetOptions{})
+		existingSvc, err := c.context.Clientset.CoreV1().Services(cephObjectStore.Namespace).Get(instanceName(cephObjectStore.Name), metav1.GetOptions{})
 		if err != nil {
 			return "", errors.Wrapf(err, "failed to get existing service IP")
 		}
-		return svc.Spec.ClusterIP, nil
+		// copy necessary properties from the existing service for the update
+		service.Spec.ClusterIP = existingSvc.Spec.ClusterIP
+		service.ObjectMeta = existingSvc.ObjectMeta
```

Comment:
Why is the object meta a necessary here?

Reply:
When updating the service, K8s complained about two requirements for updating the svc: 
- The `ClusterIP` must be set (instead of blank) since it's immutable after creation
- The `resourceVersion` must be set. Copying the metadata might be overkill for this, but I didn't want to risk some other required property being missed in the update.

Reply:
Hmm... 

In `pkg/operator/k8sutil/service.go` there is already a `CreateOrUpdateService` method that does the `ClusterIP` and `ResourceVersion` updates. What if we use that?

Reply:
Right, good catch!